### ISSUE TITLE
Should the prototypes use ksonnet-lib more sparingly?

### ISSUE THREAD

**Issue by hausdorff** [ksonnet/ksonnet#209]
We are getting some feedback in slack that indicates prodigious use of ksonnet-lib is intimidating to people who don't know Jsonnet. We use ksonnet-lib throughout the parts prototypes to make it possible to "mix in" volume mounts in a container list, but it is worth wondering if we should try to make the prototype code itself as familiar as possible.

So, taking redis as an example, instead of something like:

```
std.prune(k.core.v1.list.new([
  redis.parts.deployment.persistent(name, secretName),
  redis.parts.pvc(name),
  redis.parts.svc.metricDisabled(name),
  optionalSecret,
]))
```

perhaps something like:

```
{
  kind: "List",
  apiVersion: "v1",
  items: [
    redis.parts.deployment.persistent(name, secretName),
    redis.parts.pvc(name),
    redis.parts.svc.metricDisabled(name),
    optionalSecret,
  ],
}
```

The latter has the effect of using Jsonnet language features as they're needed.

Thoughts? cc @jbeda @jessicayuen @jessicayuen @bryanl

**Comment by bryanl** [comment]
Perhaps we need a stdlib function that provides some additional sugar over the jsonnet stdlib? `std.prune` and `k.core.v1.list.new` could be a bit opaque for the uninitiated.

**Comment by hausdorff** [comment]
@bryanl I'm not sure I understand -- I'm proposing that using ksonnet-lib prolifically in prototypes makes it more intimidating to users. To me the solution is to have _less_ sugar rather than more, _i.e._, making it look more like JSON, rather than less. Not sure if that makes sense.

**Comment by jbeda** [comment]
I agree with @hausdorff.  The power of jsonnet is that it looks mostly like json with some programmable features. The way we have it now it looks nothing like json.  Would be interesting to find ways to make it a bit less opaque.

**Comment by redbaron** [comment]
#209 is this very issue :)  Follow up is in https://github.com/ksonnet/ksonnet/issues/548
